src/org/zaproxy/zap/extension/exportreport/Export/ReportExport.java

```diff
@@ -0,0 +1,315 @@
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ * This file is based on the Paros code file ReportLastScan.java
+ */
+package org.zaproxy.zap.extension.exportreport.Export;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.io.Writer;
+import java.nio.channels.Channels;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.TransformerFactoryConfigurationError;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.log4j.Logger;
+
+import org.json.*;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.extension.ViewDelegate;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.SiteMap;
+import org.parosproxy.paros.model.SiteNode;
+import org.zaproxy.zap.extension.exportreport.FileChooser.Utils;
+
+import org.zaproxy.zap.extension.exportreport.model.AlertItem;
+import org.zaproxy.zap.extension.exportreport.model.Alerts;
+import org.zaproxy.zap.extension.exportreport.model.Report;
+import org.zaproxy.zap.extension.exportreport.model.Sites;
+import org.zaproxy.zap.utils.XMLStringUtil;
+import org.zaproxy.zap.view.ScanPanel;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.ls.DOMImplementationLS;
+import org.w3c.dom.ls.LSSerializer;
+import org.xml.sax.SAXException;
+
+/*
+ * AUTHOR		:	GORAN SARENKAPA - JordanGS
+ * SPONSOR		:	RYERSON UNIVERSITY
+ */
+
+public class ReportExport {
+	
+	private static Logger logger = Logger.getLogger(ExportReport.class);
+	
+	private static String entityEncode(String text) throws UnsupportedEncodingException {
+		String result = text;
+
+		if (result == null) {
+			return result;
+		}
+		// There is an encoding issue with the passed in String, this is a fix to maintain encoding and escapes!
+		byte ptext[] = result.getBytes("ISO-8859-1"); 
```

Comment:
- Which encoding issues were you seeing?
 - Could use StandardCharsets.ISO_8859_1 instead.